Sources/DeepLearning/MNISTdataset.swift

```diff
@@ -0,0 +1,91 @@
+import Python
+import TensorFlow
+
+let np = Python.import("numpy")
+let path = Python.import("os.path")
+let urllib = Python.import("urllib.request")
+let gzip = Python.import("gzip")
+let sh = Python.import("shutil")
+
+let URL = "http://yann.lecun.com/exdb/mnist/"
+let trainImagesFile = "train-images-idx3-ubyte.gz"
+let trainLabelsFile = "train-labels-idx1-ubyte.gz"
+let testImagesFile = "t10k-images-idx3-ubyte.gz"
+let testLabelsFile = "t10k-labels-idx1-ubyte.gz"
+let trainImages = "train-images-idx3-ubyte"
+let trainLabels = "train-labels-idx1-ubyte"
+let testImages = "t10k-images-idx3-ubyte"
+let testLabels = "t10k-label-idx1-ubyte"
+let imageSize = 28
+let trainCount = 60000
+let testCount = 10000
+
+// Function to download and extract from online
+func download_and_extract(){
```

Comment:
To ensure consistency and readability, please use camel case for all names in Swift, including function names and variable names.

All our library code is following the [Google Swift Style](https://google.github.io/swift) with the exception that indentation is 4 spaces long. It’d be great if you take a look!